ramalama/sandbox.py

```
@@ -180,9 +222,108 @@ class Agent:
+
+            if not os.path.isfile(tarpath):
+                perror(f"{kind_key[:-1].capitalize()} artifact not found: {name} -> {tarpath}")
+                continue
+
+            try:
+                with tarfile.open(tarpath, 'r:gz') as tar:
+                    # extract into a directory matching the same slug used for the
+                    # mount destination above, under this kind's own subdirectory
+                    extract_dir = os.path.join(tmpdir, kind_info["dest"], slug)
+                    os.makedirs(extract_dir, exist_ok=True)
+                    tar.extractall(path=extract_dir)
```

Ruff (0.16.1)

[error] Uses of `tarfile.extractall()`

(S202)